presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -806,6 +806,13 @@ protected Scope visitJoin(Join node, Scope scope)
 
                 addCoercionForJoinCriteria(node, leftExpression, rightExpression);
                 expressions.add(new ComparisonExpression(EQUAL, leftExpression, rightExpression));
+
+                if (node.getType() == Join.Type.LEFT) {
+                    output = createScope(node, scope, left.getRelationType().joinWith(right.hide(rightExpression).getRelationType()));
```

Comment:
Although I'm not sure if this is correct:
```
dphillips=# SELECT a.*, b.* FROM (VALUES (1), (2)) a (x) FULL JOIN (VALUES (1), (3)) b (x) USING (x);
 x | x 
---+---
 1 | 1
 2 |  
   | 3
(3 rows)
```
I think they should not be resolved as part of `a.*`, although it is useful if they are available as a hidden column.